interpreter/terminal_interface/terminal_interface.py

```diff
@@ -17,8 +18,9 @@ def terminal_interface(interpreter, message):
             "**Open Interpreter** will require approval before running code."
         ]
 
-        if interpreter.safe_mode != "off":
-            interpreter_intro_message.append(f"**Safe Mode**: {interpreter.safe_mode}\n\n>Note: **Safe Mode** requires `semgrep` (`pip install semgrep`)")
+        if interpreter.safe_mode == "ask" or interpreter.safe_mode == "auto":
+            if not check_for_package("semgrep"):
+                interpreter_intro_message.append(f"**Safe Mode**: {interpreter.safe_mode}\n\n>Note: **Safe Mode** requires `semgrep` (`pip install semgrep`)")
```

Comment:
Now we only need to show the `semgrep` message if it isn't already in the user's current environment. If `semgrep` is available and `safe_mode` is enabled, `semgrep` will be automatically loaded.